src/pages/SearchPage/SearchPage.js

```diff
@@ -6,21 +6,22 @@ import Spinner from '../../components/Spinner/Spinner';
 
 function SearchPage() {
   const history = useHistory();
-
   const [isLoading, setIsLoading] = useState(false);
 
-  async function fetchData() {
+  async function fetchData(searchTerm) {
+    const timeNow = new Date();
+    const timeLastYear = Math.round(timeNow.getTime() / 1000) - 31536000; // Seconds in year
     setIsLoading(true);
     const response = await fetch(
-      'https://www.reddit.com/r/javascript/search.json?q=oop&limit=5',
+      `https://api.pushshift.io/reddit/search/submission/?subreddit=${searchTerm}&sort=desc&sort_type=created_utc&after=${timeLastYear}&size=500`,
```

Comment:
Haha, you did a good research. I didn't find this API. I used `https://www.reddit.com/r/javascript/top.json` which is more complicated since you have to send 5 requests that depend on each other. So well done